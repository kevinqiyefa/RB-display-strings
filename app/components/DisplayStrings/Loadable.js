/**
 *
 * Asynchronously loads the component for DisplayStrings
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
