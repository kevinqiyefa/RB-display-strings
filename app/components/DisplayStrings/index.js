/**
 *
 * DisplayStrings
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

function DisplayStrings(props) {
  const strings = props.strings.map(str => <li key={str.id}>{str.string}</li>);
  return <ul>{strings}</ul>;
}

DisplayStrings.propTypes = {
  strings: PropTypes.array,
};

export default memo(DisplayStrings);
