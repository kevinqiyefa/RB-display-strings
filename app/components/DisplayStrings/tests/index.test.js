/**
 *
 * Tests for DisplayStrings
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import DisplayStrings from '../index';

describe('<DisplayStrings />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<DisplayStrings strings={[]} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to render ', () => {
    const { getByText } = render(
      <DisplayStrings strings={[{ id: 1, string: 'testing' }]} />,
    );
    expect(getByText('testing')).toBeTruthy();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<DisplayStrings strings={[]} />);

    expect(firstChild).toMatchSnapshot();
  });
});
