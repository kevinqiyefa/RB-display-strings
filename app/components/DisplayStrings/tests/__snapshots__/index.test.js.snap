// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayStrings /> Should render and match the snapshot 1`] = `
<div
  class="ContentWrapper-sc-1pl7b9y-0 kNVNXH"
>
  <h1
    class="DisplayStrings__ContentHeader-sc-1743nj8-0 jZIUMu"
  >
    Strings From Database:
  </h1>
  <ul
    class="UL-sc-16unqrb-0 PCfYS"
  />
</div>
`;
